lib/icinga/downtime.cpp

```diff
@@ -427,6 +442,34 @@ bool Downtime::CanBeTriggered()
 	return true;
 }
 
+void Downtime::SetupCleanupTimer()
+{
+	double deadline;
+
+	if (GetFixed()) {
+		deadline = GetEndTime();
+	} else {
+		auto triggerTime (GetTriggerTime());
+
+		if (triggerTime <= 0) {
+			return;
+		}
+
+		deadline = triggerTime + GetDuration();
+	}
```

Comment:
That's different logic to compute then end compared to `Downtime::IsExpired()`. Did you test whether your code handles flexible downtimes that never triggered correctly?